src/test/scala/cognite/spark/v1/FileContentRelationTest.scala

```diff
@@ -0,0 +1,248 @@
+        .option("externalId", fileWithWrongMimeTypeExternalId)
+        .load()
+      sourceDf.createOrReplaceTempView("fileContent")
+      spark.sqlContext.sql(s"select * from filecontent").collect()
+    }
+    assert(exception.getMessage.contains("Wrong mimetype. Expects application/jsonlines"))
+  }
+
+  it should "infer the schema" in {
+    val sourceDf: DataFrame = dataFrameReaderUsingOidc
+      .useOIDCWrite
+      .option("type", "filecontent")
+      .option("inferSchema", value = true)
+      .option("externalId", fileExternalId)
+      .load()
+    sourceDf.createOrReplaceTempView("fileContent")
+    sourceDf.schema.fields should contain allElementsOf Seq(
+      StructField("name", StringType, nullable = true),
+      StructField("test", StringType, nullable = true),
+      StructField("age", LongType, nullable = true)
+    )
+  }
+
+  it should "infer the schema correctly with nested json" in {
+    val sourceDf: DataFrame = dataFrameReaderUsingOidc
+      .useOIDCWrite
+      .option("type", "filecontent")
+      .option("inferSchema", value = true)
+      .option("externalId", fileExternalIdWithNestedJson)
+      .load()
+    sourceDf.createOrReplaceTempView("fileContent")
+    sourceDf.schema.fields should contain allElementsOf Seq(
+      StructField("name", StringType, nullable = true),
+      StructField("test",
+        ArrayType(
+          StructType(
+            Seq(
+              StructField("key",StringType, nullable = true),
+              StructField("key2",StringType, nullable = true)
+            )
+          ),
+          containsNull = true
+        ),
+        nullable = true
+      ),
+      StructField("age", LongType, nullable = true)
+    )
+  }
+
+  it should "infer the schema correctly with conflicting types" in {
+    val sourceDf: DataFrame = dataFrameReaderUsingOidc
+      .useOIDCWrite
+      .option("type", "filecontent")
+      .option("inferSchema", value = true)
+      .option("externalId", fileExternalIdWithConflicts)
+      .load()
+    sourceDf.createOrReplaceTempView("fileContent")
+    sourceDf.schema.fields should contain allElementsOf Seq(
+      StructField("name", StringType, nullable = true),
+      StructField("test", StringType, nullable = true),
+      StructField("age", LongType, nullable = true)
+    )
+  }
+
+
+  it should "select specific columns" in {
+    val sourceDf: DataFrame = dataFrameReaderUsingOidc
+      .useOIDCWrite
+      .option("type", "filecontent")
+      .option("externalId", fileExternalId)
+      .option("inferSchema", true)
+      .load()
+    sourceDf.createOrReplaceTempView("fileContent")
+    val result = spark.sqlContext.sql(s"select name, test from filecontent").collect()
+    result.map(_.toSeq.toList) should contain theSameElementsAs
+      Seq(
+        List("Alice", null),
+        List("Bob", null),
+        List("Charlie", null),
+        List("Charlie2", "test")
+      )
+  }
+
+  it should "not infer schema is not asked to" in {
```

Comment:
assertion on resulting schema is missing? Should still be able to check that it is "json"

Reply:
added